UnitTests/ConsoleDrivers/KeyEventArgsTests.cs

```diff
@@ -0,0 +1,238 @@
+	public void IsAlpha (Key key, bool expected)
+	{
+		var eventArgs = new KeyEventArgs (key);
+		Assert.Equal (expected, eventArgs.IsAlpha);
+	}
+
+	[Theory]
+	[InlineData ((Key)'❿', '❿')]
+	[InlineData ((Key)'☑', '☑')]
+	[InlineData ((Key)'英', '英')]
+	[InlineData ((Key)'{', '{')]
+	[InlineData ((Key)'\'', '\'')]
+	[InlineData ((Key)'\r', '\r')]
+	[InlineData ((Key)'ó', 'ó')]
+	[InlineData ((Key)'ó' | Key.ShiftMask, 'ó')]
+	[InlineData ((Key)'Ó', 'Ó')]
+	[InlineData ((Key)'ç' | Key.ShiftMask | Key.AltMask | Key.CtrlMask, default)]
+
+	[InlineData ((Key)'a', 97)] // 97 or Key.Space | Key.A
+	[InlineData ((Key)'A', 97)] // 65 or equivalent to Key.A, but A-Z are mapped to lower case by drivers
+	//[InlineData (Key.A, 97)] // 65 equivalent to (Key)'A', but A-Z are mapped to lower case by drivers
+	[InlineData (Key.ShiftMask | Key.A, 65)]
+	[InlineData (Key.CtrlMask | Key.A, default)]
+	[InlineData (Key.AltMask | Key.A, default)]
+	[InlineData (Key.ShiftMask | Key.CtrlMask | Key.A, default)]
+	[InlineData (Key.ShiftMask | Key.AltMask | Key.A, default)]
+	[InlineData (Key.AltMask | Key.CtrlMask | Key.A, default)]
+	[InlineData (Key.ShiftMask | Key.CtrlMask | Key.AltMask | Key.A, default)]
+	
+	[InlineData ((Key)'z', 'z')]
+	[InlineData ((Key)'Z', 'z')]
+	[InlineData (Key.ShiftMask | Key.Z, 'Z')]
+
+	[InlineData ((Key)'1', '1')]
+	[InlineData (Key.ShiftMask | Key.D1, '1')]
+	[InlineData (Key.CtrlMask | Key.D1, default)]
+	[InlineData (Key.ShiftMask | Key.CtrlMask | Key.D1, default)]
+
+	[InlineData (Key.F1, default)]
+	[InlineData (Key.ShiftMask | Key.F1, default)]
+	[InlineData (Key.CtrlMask | Key.F1, default)]
+
+	[InlineData (Key.Enter, '\n')]
+	[InlineData (Key.Tab, '\t')]
+	[InlineData (Key.Esc, 0x1b)]
+	[InlineData (Key.Space, ' ')]
+
+	[InlineData (Key.ShiftMask | Key.CtrlMask | Key.AltMask | Key.Enter, default)]
+
+	[InlineData (Key.Null, default)]
+	[InlineData (Key.ShiftMask | Key.CtrlMask | Key.AltMask | Key.Null, default)]
+
+	[InlineData (Key.CharMask, default)]
+	[InlineData (Key.SpecialMask, default)]
+	public void AsRune_ShouldReturnCorrectIntValue (Key key, Rune expected)
+	{
+		var eventArgs = new KeyEventArgs (key);
+		Assert.Equal (expected, eventArgs.AsRune);
+	}
+
+	[Theory]
+	[InlineData (Key.AltMask, true)]
+	[InlineData (Key.A, false)]
+	public void IsAlt_ShouldReturnCorrectValue (Key key, bool expected)
+	{
+		var eventArgs = new KeyEventArgs (key);
+		Assert.Equal (expected, eventArgs.IsAlt);
+	}
+
+	// Similar tests for IsShift and IsCtrl
+	[Fact]
+	public void ToString_ShouldReturnReadableString ()
+	{
+		var eventArgs = new KeyEventArgs (Key.CtrlMask | Key.A);
+		Assert.Equal ("Ctrl+A", eventArgs.ToString ());
+	}
+
+	[Theory]
+	[InlineData (Key.CtrlMask | Key.A, '+', "Ctrl+A")]
+	[InlineData (Key.AltMask | Key.B, '-', "Alt-B")]
+	public void ToStringWithSeparator_ShouldReturnFormattedString (Key key, Rune separator, string expected)
+	{
+		Assert.Equal (expected, KeyEventArgs.ToString (key, separator));
+	}
```

Comment:
You have to change the separator to `char`, otherwise we can't set a breakpoint because it won't stop on it.

```cs
	public void ToStringWithSeparator_ShouldReturnFormattedString (Key key, char separator, string expected)
	{
		Assert.Equal (expected, KeyEventArgs.ToString (key, (Rune)separator));
	}

```